internal/config/nsmgr/nsmgr.go

```diff
@@ -0,0 +1,157 @@
+package nsmgr
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	nspkg "github.com/containernetworking/plugins/pkg/ns"
+	"github.com/containers/storage/pkg/idtools"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+)
+
+// NamespaceManager manages the server's namespaces.
+// Specifically, it is an interface for how the server is creating namespaces.
+// and can be requested to create namespaces for a pod.
+type NamespaceManager interface {
+	// NewPodNamespaces creates new namespaces for a pod.
+	NewPodNamespaces(managedNamespaces []NSType, idMappings *idtools.IDMappings, sysctls map[string]string) ([]Namespace, error)
+}
+
+// New creates a new NamespaceManager.
+func New(namespacesDir, pinnsPath string) NamespaceManager {
+	return &namespaceManager{
+		namespacesDir: namespacesDir,
+		pinnsPath:     pinnsPath,
+	}
+}
+
+// namespaceManager is the internal implementation for NamespaceManager.
+type namespaceManager struct {
+	namespacesDir string
+	pinnsPath     string
+}
+
+// NewPodNamespaces creates new namespaces for a pod.
+// It's responsible for running pinns and creating the Namespace objects.
+// The caller is responsible for cleaning up the namespaces by calling Namespace.Remove().
+func (mgr *namespaceManager) NewPodNamespaces(managedNamespaces []NSType, idMappings *idtools.IDMappings, sysctls map[string]string) ([]Namespace, error) {
+	if len(managedNamespaces) == 0 {
+		return []Namespace{}, nil
```

Comment:
AFAIK this can be `return nil, nil` (no need to create an empty slice).

Reply:
It might not be good though (the caller would need an additional `!= nil` check), so ignore this.

Reply:
yeah I am worried about needing the nil check, so I'd rather just make a new one here